src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -331,6 +349,27 @@ async Task<InProgressState> CollapseInProgressStateAsync(InProgressState initial
                     try
                     {
                         var currentState = initialState;
+
+                        // To speed things up, we look for all the added documents in the chain and we preemptively kick
+                        // off work to parse the documents for it in parallel.  This has the added benefit that if
+                        // someone asks for a frozen partial snapshot while we're in the middle of doing this, they can
+                        // use however many document states have successfully parsed their syntax trees.  For example,
+                        // if you had one extremely large file that took a long time to parse, and dozens of tiny ones,
+                        // it's more likely that the frozen tree would have many more documents in it.
+                        var documentCount = currentState.PendingTranslationActions.Sum(
+                            a => a is TranslationAction.AddDocumentsAction { Documents: var documents } ? documents.Length : 0);
+                        using var _ = ArrayBuilder<Task>.GetInstance(documentCount, out var parsingTasks);
+
+                        foreach (var action in currentState.PendingTranslationActions)
+                        {
+                            if (action is TranslationAction.AddDocumentsAction { Documents: var documents })
+                            {
+                                foreach (var document in documents)
+                                    parsingTasks.Add(Task.Run(async () => await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false), cancellationToken));
```

Comment:
Are we adding things to this list only to not use them? I get fire and forget feels weird, but this is just fire and forget with more allocations.

Reply:
(this PR otherwise seems fine but I'm not sure if this is an oversight or something)

Reply:
You're correct. It serves no purpose.  I will remove :-)